### ISSUE TITLE
[BUG] Query not working anymore with Memgraph 2.9

### ISSUE THREAD

**Issue by dgarros** [memgraph/memgraph#1100]
**Memgraph version**
2.9

**Environment**
Memgraph 2.9 running in Docker, storage mode IN_MEMORY_ANALYTICAL
Python 3.9
Neo4j driver for Python using the Bolt interface (tested with v5.7 and v5.10)

<details>
  <summary>Memgraph Configuration</summary>

```
--bolt-server-name-for-init=Neo4j/
--log-file=/var/log/memgraph/memgraph.log
--log-level=WARNING
--memory-limit=0
--metrics-address=0.0.0.0
--metrics-port=2004
--query-modules-directory=/usr/lib/memgraph/query_modules
--storage-parallel-index-recovery=false
--storage-properties-on-edges=true
--storage-recover-on-startup=true
--storage-snapshot-interval-sec=300
--storage-snapshot-on-exit=true
--storage-snapshot-retention-count=3
--storage-wal-enabled=true
--data-directory=/var/lib/memgraph
--isolation-level=READ_COMMITTED
--telemetry-enabled=true
```

</details>

**Describe the bug**
When we execute the following query in Memgraph 2.9, the python driver is complaining that the data returned is not valid

```cypher
MATCH p = (s)-[:IS_RELATED*..2]-(d)
WHERE ID(s) = 11028 AND ID(d) = 11040
RETURN p
```

```python 
self = <Record p=<neo4j._codec.hydration._common.BrokenHydrationObject object at 0x10b3c0fa0>>, index = 0

    def _super_getitem_single(self, index):
        value = super().__getitem__(index)
        if isinstance(value, BrokenHydrationObject):
>           raise self._broken_record_error(index) from value.error
E           neo4j.exceptions.BrokenRecordError: Record contains broken data at 0 ('p')
```

**To Reproduce**
1. Run the following query
```cypher
MATCH p = (s)-[:IS_RELATED*..2]-(d)
WHERE ID(s) = 11028 AND ID(d) = 11040
RETURN p
```

**Expected behavior**
The exact same code/query is working fine with Memgraph 2.8
When I run the same query on Memgraph-lab, it seems to work as well

**Logs**

**Additional context**
Here is the result when i execute the same query from Memgraph-lab 
```json
{
    "p": {
        "nodes": [
            {
                "id": 11028,
                "labels": [
                    "Node",
                    "BuiltinTag"
                ],
                "properties": {
                    "kind": "BuiltinTag",
                    "uuid": "fb2406d6-dd44-4216-ae09-f8e89377e2e2"
                },
                "type": "node"
            },
            {
                "id": 11045,
                "labels": [
                    "Relationship"
                ],
                "properties": {
                    "name": "builtintag__testperson",
                    "uuid": "9a551ed0-933c-4e01-8fb2-179dcc87e81a"
                },
                "type": "node"
            },
            {
                "id": 11040,
                "labels": [
                    "Node",
                    "TestPerson"
                ],
                "properties": {
                    "kind": "TestPerson",
                    "uuid": "705badb3-661f-4b70-9f14-2bcd9ec6aa45"
                },
                "type": "node"
            }
        ],
        "relationships": [
            {
                "id": 36897,
                "start": 11028,
                "end": 11045,
                "label": "IS_RELATED",
                "properties": {
                    "branch": "main",
                    "branch_level": 1,
                    "from": "2023-07-25T05:16:44.370399Z",
                    "status": "active"
                },
                "type": "relationship"
            },
            {
                "id": 36896,
                "start": 11040,
                "end": 11045,
                "label": "IS_RELATED",
                "properties": {
                    "branch": "main",
                    "branch_level": 1,
                    "from": "2023-07-25T05:16:44.370399Z",
                    "status": "active"
                },
                "type": "relationship"
            }
        ],
        "type": "path"
    }
}
```

**Comment by gitbuda** [comment]
Hi @dgarros! In v2.9 we added the support for Bolt v5, it might be that there is some incompatibility... figuring out what's the problem 👀

**Comment by gitbuda** [comment]
The impact here is high because, most likely, a lot of Python (at least, probably other language drivers) use cases might not work with v5 of Bolt drivers.

@dgarros is it an option for you to use v4 of the Python driver? (that might work, I haven't tested it by myself)

**Comment by dgarros** [comment]
@gitbuda I'm happy to do more testing but I'm not sure how I can force it to use Bolt v4 instead of v5
is that something I can configure on memgraph ?

I looked quickly at the driver documentation and at Memgraph documentation and I couldn't find it on either side.

**Comment by gitbuda** [comment]
@dgarros we managed to reproduce + fix the issue (hopefully, the PR is linked) 💪

To use different versions of Bolt, just use different client versions installed via e.g. pip, server and client have an automatic resolution of which version is going to be used (based on client version and what server supports)

How critical is it for you to have this released this week? We plan to release v2.10 next week with this fix included. Does that work for you?

**Comment by dgarros** [comment]
> To use different versions of Bolt, just use different client versions installed via e.g. pip, server and client have an automatic resolution of which version is going to be used (based on client version and what server supports)

Got it, thanks for the info

> How critical is it for you to have this released this week? We plan to release v2.10 next week with this fix included. Does that work for you?

That's perfect for me, I can wait
thanks for the quick fix
